control-plane/api/v1alpha1/servicedefaults_types.go

```diff
@@ -492,3 +518,49 @@ func (in *ServiceDefaults) MatchesConsul(candidate capi.ConfigEntry) bool {
 func (in *ServiceDefaults) ConsulGlobalResource() bool {
 	return false
 }
+
+func (in EnvoyExtensions) toConsul() []capi.EnvoyExtension {
+	if in == nil {
+		return nil
+	}
+
+	outConfig := make([]capi.EnvoyExtension, 0)
+
+	for _, e := range in {
+		consulExtension := capi.EnvoyExtension{
+			Name:     e.Name,
+			Required: e.Required,
+		}
+
+		// We already validate that arguments is present
+		var args map[string]interface{}
+		_ = json.Unmarshal(e.Arguments, &args)
+		consulExtension.Arguments = args
+		outConfig = append(outConfig, consulExtension)
+	}
+
+	return outConfig
+}
+
+func (in EnvoyExtensions) validate(path *field.Path) field.ErrorList {
+	if len(in) == 0 {
+		return nil
+	}
+
+	var errs field.ErrorList
+	for i, e := range in {
+		if err := e.validate(path.Child("envoyExtension").Index(i)); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (in EnvoyExtension) validate(path *field.Path) *field.Error {
+	if in.Arguments == nil {
```

Comment:
We should validate that the arguments provided are json parsable similar to https://github.com/hashicorp/consul-k8s/blob/7fa7336b6bd002efed5fea88990c3f6d2f62d94f/control-plane/api/v1alpha1/proxydefaults_types.go#L236